core/src/main/java/nu/yona/server/analysis/entities/IntervalActivity.java

```diff
@@ -98,7 +99,7 @@ public UserAnonymized getUserAnonymized()
 
 	public Goal getGoal()
 	{
-		return goal;
+		return EntityUtil.enforceLoading(goal);
```

Comment:
Maybe add a comment "this is necessary because instances of this class are cached outside the hibernate session".

I guess this has to be done for all entity object references that somehow end up in the cache. Maybe we can introduce an interface for all cached objects with a method `forceLoadReferedEntityObjects` that is called just after computing them (or alternatively do some complicated things with reflection). This will become quite complicated however...